core/src/gl/shaderProgram.cpp

```diff
@@ -84,15 +120,20 @@ GLint ShaderProgram::getAttribLocation(const std::string& _attribName) {
 
 }
 
-GLint ShaderProgram::getUniformLocation(const std::string& _uniformName) {
+GLint ShaderProgram::getUniformLocation(const UniformEntries::UniformEntry* _entry) {
```

Comment:
Yes, this would also be my suggestion or to even cache the uniform value on the caller side. Though lights are a special (or more complicated) case where I got stuck when I tried this 'uniform caching' a while ago.